crud/common/utils.lua

```diff
@@ -96,8 +96,32 @@ function utils.format_replicaset_error(replicaset_uuid, msg, ...)
     )
 end
 
-function utils.get_space(space_name, replicasets)
-    local replicaset = select(2, next(replicasets))
+function utils.get_space(space_name, vshard_router)
+    local replicasets, replicaset
+    local deadline = fiber.clock() + const.GET_SPACE_TIMEOUT
+    repeat
+        -- Try to get master with timeout.
+        fiber.yield()
+        replicasets = vshard_router:routeall()
+        replicaset = select(2, next(replicasets))
+        if replicaset ~= nil and
+           replicaset.master ~= nil and
+           replicaset.master.conn.error == nil then
+            -- Break if there is master in replicaset and
+            -- connection to master is established.
+            break
+        end
+    until (
+        -- If at least one of the lower conditions in this block is not met
+        -- and the deadline has not yet been reached, the cycle will be repeated.
+        replicaset ~= nil and
```

Comment:
Well, I don't see a reason to do double-check here. For now there is no cases when the first condition (before `or`) may stop the cycle.

Since we want to try to extract replicasets at least once, `do ... while` seems more suitable here.

Reply:
Done, simplified 

AFAIS, the syntactic construction `do ... while (...)` dose not exist in Lua, and in similar cases we can use `repeat ... until NOT (...)`. 
Now I have rewritten the loop with `while (...) do ... end`, and as far as I understand, we can be sure that it will be run at least once (only if there is no anomaly with the `fiber.clock()` call inside while condition block)

Reply:
> AFAIS, the syntactic construction `do ... while (...)` dose not exist in Lua, and in similar cases we can use `repeat ... until NOT (...)`.

Yeah, sorry, I was misled by some Lua tutorial website yesterday evening.